src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -702,11 +702,13 @@ public bool IsReadOnly
         public void Assign(string expression)
         {
             this.VerifyNotDisposed();
+            int threadId = Client.GetDebuggedThread().Id;
+            uint frameLevel = _ctx.Level;
```

Comment:
Should these two lines be moved within the lambda passed to `RunOperation` to avoid race conditions that change the `threadId` or `frameLevel`?

I think I see why a `continue` could also cause a race condition that messes up this operation. Not sure if a continue would automatically dispose these `VariableInformation` object (I couldn't find where).

However, I expect the same problem to occur for `VariableInformation::Eval`, which for some reason doesn't use the `_engine.DebuggedProcess.WorkerThread.RunOperation pattern`. I think the worst that can happen is that `threadId` and `frameLevel` are not valid anymore and the resulting expression evaluation will fail.

Do you agree?